pkg/promlib/client/client.go

```diff
@@ -24,22 +24,29 @@ type doer interface {
 // objects, we have to go through them and then serialize again into DataFrame which isn't very efficient. Using custom
 // client we can parse response directly into DataFrame.
 type Client struct {
-	doer    doer
-	method  string
-	baseUrl string
+	doer         doer
+	method       string
+	baseUrl      string
+	queryTimeout string
 }
 
-func NewClient(d doer, method, baseUrl string) *Client {
-	return &Client{doer: d, method: method, baseUrl: baseUrl}
+const defaultQueryTimeout = "60s"
+
+func NewClient(d doer, method, baseUrl, queryTimeout string) *Client {
+	if queryTimeout != "" {
```

Comment:
Why are we overwriting the passed `queryTimeout`? Should not this be `queryTimeOut == ""`?

Reply:
Yes, you're probably right. Question is if it should default to `60s` or not set/passed when empty. Might be a breaking change (or unexpected behavior) for those that haven't configured either timeout or query timeout.

Reply:
@s4kh @marefr  You are right it is a mistake. Must be `==`.
@marefr That's a good point. if that's not specified we shouldn't use it.